pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -151,15 +151,19 @@ class CreateVM extends React.Component {
             break;
         case 'sourceType':
             this.setState({ [key]: value });
-            if (value != PXE_SOURCE) {
-                notifyValuesChanged('source', null);
-            } else {
+            if (value == PXE_SOURCE) {
                 let initialPXESource = getPXEInitialNetworkSource(this.props.nodeDevices,
                                                                   this.props.networks,
                                                                   this.props.vmParams.connectionName);
 
                 notifyValuesChanged('source', initialPXESource);
             }
+            // Reset the source when the previous selection was PXE;
+            // all the other choices are string set by the user
+            else if (this.state.sourceType == PXE_SOURCE && value != PXE_SOURCE) {
+                notifyValuesChanged('source', '');
```

Comment:
This is very confusing but it should remain here due to bad design of the createVm component. I will work today on seperate PR to fix it.